### ISSUE TITLE
E: Unable to locate package libatlas3-base

### ISSUE THREAD

**Issue by jeffhammond** [travis-ci/apt-package-whitelist#2159]
https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise#L3762 shows `libatlas3-base` but when I try to use it (via https://github.com/m-a-d-n-e-s-s/madness/blob/340bfed7bdeb3815c3e68e3aa1e660db57a6a6e9/.travis.yml), I get this error:
```sh
Installing APT Packages (BETA)
$ export DEBIAN_FRONTEND=noninteractive
11.99s$ sudo -E apt-get -yq update &>> ~/apt-get-update.log
3.48s$ sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install gcc-4.7 g++-4.7 gfortran-4.7 gcc-4.8 g++-4.8 gfortran-4.8 gcc-4.9 g++-4.9 gfortran-4.9 gcc-5 g++-5 gfortran-5 clang-3.6 libblas-dev liblapack-dev libopenblas-base libatlas3-base libgoogle-perftools-dev libtbb-dev
Reading package lists...
Building dependency tree...
Reading state information...
E: Unable to locate package libatlas3-base
```

As shown on https://travis-ci.org/m-a-d-n-e-s-s/madness/builds/99759047, this broke every one of my otherwise mostly working builds.

**Comment by ax3l** [comment]
that's because `libatlas3-base` was first introduced [in trusty](http://packages.ubuntu.com/trusty/libatlas3-base), not in precise.

**Comment by jeffhammond** [comment]
Sure, but since the file in question is named **ubuntu-precise**, it is reasonable for one to conclude that the contents apply to Ubuntu Precise and not Ubuntu Trusty, no?

**Comment by ax3l** [comment]
I think they will/do simply reuse it for the trusty image right now, too.

nevertheless, this repo is [mainly maintained by a bot](https://github.com/travis-ci/apt-package-whitelist#package-approval-process), don't expect official answers ;)

**Comment by BanzaiMan** [comment]
There is something amiss, to be sure.

See https://github.com/travis-ci/apt-package-whitelist/pull/1858 and its test https://travis-ci.org/travis-ci/apt-whitelist-checker/builds/93211736. The source package was found in "precise/universe" (https://travis-ci.org/travis-ci/apt-whitelist-checker/builds/93211736#L549). I think this source should exist on the container image; if it does and the package is not found, then maybe `apt-get update` failed. Try dumping the content of the log file: `cat ~/apt-get-update.log`.

**Comment by ax3l** [comment]
I think the [*source* package](http://packages.ubuntu.com/source/trusty/atlas) ([right hand side on this page](http://packages.ubuntu.com/trusty/libatlas3-base)) is simply called `atlas` which *creates* the `deb` packages `Binary: libatlas3-base, libatlas3gf-base, libatlas-base-dev, libatlas-dev, libatlas-test, libatlas-doc` (see [`.dsc` file](http://archive.ubuntu.com/ubuntu/pool/universe/a/atlas/atlas_3.10.1-4.dsc) of the source package).

The source package `atlas` exists in both precise & trusty.

**Comment by ax3l** [comment]
@jeffhammond look at the difference between [trusty](http://packages.ubuntu.com/source/trusty/atlas) and [precise](http://packages.ubuntu.com/source/precise/atlas): the package `libatlas3-base` was only introduced in the newer `trusty` but you can still use the [transitional package `libatlas3gf-base`](http://packages.ubuntu.com/trusty/libatlas3gf-base) that already existed in `precise` for the same purpose.

**Comment by jeffhammond** [comment]
@ax3l I don't want to have to cross-check Travis whitelists with third-party Ubuntu sites every time I consider a package.  I want a file named `${OPERATING_SYSTEM_DISTRIBUTION}-${SPECIFIC_VARIANT_THEREOF}` to contain information that is strictly correct, not information that is correct for `${OPERATING_SYSTEM_DISTRIBUTION}-${DIFFERENT_VARIANT_THEREOF}`.  If the whitelist files aren't going to be accurate, they should not exist.

**Comment by ax3l** [comment]
I guess it depends on your POV, but as long as I build on Ubuntu the apt repo packages.ubuntu.com of Ubuntu is my *primary* source of reference for `apt install` targets.

Totally agree, if I where them I would already split into a `ubuntu-precise` and a `ubuntu-trusty` file. Reading the README of this repo, I guess it is planned but not yet done. Also be aware that their `apt` addon is still in *beta* status.

**Comment by jeffhammond** [comment]
Yes, but Ubunutu lists packages that are _not_ whitelisted, so if I took your approach, I'd still end up checking both pages.  I'd first get the right name from Ubuntu docs and then determine if I can actually get it in Travis containers.

As it is, I am building nearly everything from source for containers anyways, which has been my response to broken packages and unreliable sources.  I get that apt container support is beta and I don't depend on it.  I reported this issue because I thought it was important to have the whitelists be correct, not because it makes Travis unusable for me.

**Comment by ax3l** [comment]
Sure, I just added content to the initial issue description so you know what's wrong in your specific case and they can make something out of this issue. I had the same problem and would prefer a list per distribution, too. Happy building.

P.S.: a better workflow as a dev might be: identify your third party dependency -> look up package names per distribution/OS as you would also recommend them that way to your users -> choose CI tool for that specific distribution -> try usual packages -> if necessary, file whitelist request.
